src/resources/transactions/transaction.ts

```diff
@@ -0,0 +1,164 @@
+import { NativeCurrencyKey } from '@/entities';
+import {
+  createQueryKey,
+  queryClient,
+  QueryFunctionArgs,
+  QueryFunctionResult,
+} from '@/react-query';
+import { useQuery } from '@tanstack/react-query';
+import {
+  consolidatedTransactionsQueryFunction,
+  consolidatedTransactionsQueryKey,
+} from './consolidatedTransactions';
+import { useAccountSettings } from '@/hooks';
+import { RainbowNetworks, getNetworkObj } from '@/networks';
+import { rainbowFetch } from '@/rainbow-fetch';
+import { ADDYS_API_KEY } from 'react-native-dotenv';
+//import { parseTransaction } from '@/parsers/transactions';
+import { Network } from '@/networks/types';
+import { RainbowError, logger } from '@/logger';
+
+type ConsolidatedTransactionsResult = QueryFunctionResult<
+  typeof consolidatedTransactionsQueryFunction
+>;
+
+export type TransactionArgs = {
+  hash: string;
+  address: string;
+  currency: NativeCurrencyKey;
+  network: Network;
+};
+
+export type BackendTransactionArgs = {
+  hash: string;
+  network: Network;
+  enabled: boolean;
+};
+
+export const transactionQueryKey = ({
+  hash,
+  address,
+  currency,
+  network,
+}: TransactionArgs) =>
+  createQueryKey(
+    'transactions',
+    { address, currency, network, hash },
+    { persisterVersion: 1 }
+  );
+
+export const fetchTransaction = async ({
+  queryKey: [{ address, currency, network, hash }],
+}: QueryFunctionArgs<typeof transactionQueryKey>) => {
+  try {
+    const chainId = getNetworkObj(network).id;
+    const url = `https://addys.p.rainbow.me/v3/${chainId}/${address}/transactions/${hash}`;
+    const response = await rainbowFetch(url, {
+      method: 'get',
+      params: {
+        currency: currency.toLowerCase(),
+      },
+      timeout: 20000,
+      headers: {
+        Authorization: `Bearer ${ADDYS_API_KEY}`,
+      },
+    });
+    const tx = response?.data?.payload?.transaction || [];
+
+    return undefined;
```

Comment:
Is this intentional?

Reply:
this is just to fix lint, so we can get this in as a smaller file change